gswikichat/vector_store_interface.py

```diff
@@ -1,24 +1,25 @@
 import os
 import json
 
-# from sentence_transformers import SentenceTransformer
 from tqdm import tqdm
 
 from haystack import Document  # , Pipeline
 from haystack.components.embedders import SentenceTransformersDocumentEmbedder
-# from haystack.components.embedders import SentenceTransformersTextEmbedder
 from haystack.document_stores.in_memory import InMemoryDocumentStore
-# from haystack.components.retrievers.in_memory import InMemoryBM25Retriever
 from haystack.components.retrievers.in_memory import InMemoryEmbeddingRetriever
-# from haystack.components.writers import DocumentWriter
 from haystack.document_stores.types.policy import DuplicatePolicy
+from haystack.components.preprocessors import DocumentSplitter
+from haystack.components.preprocessors import DocumentCleaner
 
 HUGGING_FACE_HUB_TOKEN = os.environ.get('HUGGING_FACE_HUB_TOKEN')
+EMBEDDING_CACHE_FILE = '/tmp/gbnc_embeddings.json'
```

Comment:
Do you think we can make this into an SQL library later?

We could then load only the embeddings to backend RAM, vector search the embeddings, and query the string text using the indices related to the top_k embeddings.

Reply:
Yes, I too think that we will need that optimization at some point. For the current state though, I think having this all in memory might be sufficient, simpler, and maybe even faster. I think it is reasonable to keep some hundreds MBs wikitext + embeddings in RAM, considering that the language model itself already takes some GBs. What do you think?

Reply:
@exowanderer I hit performance issues when testing our full excellent articles dataset with the json implementation. So I am experimenting with a postgres + vecto.rs implementation over here #41